sim/dex.js

```diff
@@ -132,6 +134,8 @@ class ModdedDex {
 		this.typeCache = new Map();
 		/** @type {Map<string, Effect | Move>} */
 		this.effectCache = new Map();
+		/** @type Set<string> */
+		this.callbackTypes = new Set();
```

Comment:
Out of scope for this change, but the next place I'd like to go with this optimization is to actually track `callbackTypes` by effect. From [experimentation](https://gist.githubusercontent.com/scheibo/0c925080d035216d5a1d211df8458c26/raw/fooCallbackTypes%2520(heuristic)) I've noticed that heuristically we can further constrain the types depending on the effect. ie. all `callbackTypes` are not valid for `Template`, only `onType` is used. To take that example, thats a bit more difficult because that is code in `data/statuses.js` and doesn't have an `effectType` field to know where it applies. Still, I think that's the next obvious and simple step (the final state, if necessary would be the change outlined originally in [pkmn.cc/optimize](https://pkmn.cc/optimize), but with all these changes, there might be lower hanging fruit to go after as opposed to continually chiseling away at `getRelevantEffects`).